autotyping/autotyping.py

```diff
@@ -657,3 +688,127 @@ def get_decorator_kind(dec: libcst.Decorator) -> Optional[DecoratorKind]:
+
+    containers = "deque|list|set|iterator|tuple"
+    # not using 'sequence', 'counter' or 'collection' due to likely false alarms
+
+    # (container)_(int|float|str|bool)s?
+    # e.g. list_ints => List[int]
+    # only check for built-in types to avoid false alarms, e.g. list_create, list_length
+    m = re.fullmatch(
+        rf"(?P<container>{containers})_(?P<elems>int|float|str|bool)s?", name
+    )
+    if m:  # mfw 3.7 doesn't have the walrus operator :(
+        return NamedParam("", None, m.group("elems")), [
+            m.group("container").capitalize()
+        ]
+
+    # <name>s?_(container)
+    # e.g. latitude_list => List[float]
+    # (container)_of_<name>(s)
+    # e.g. set_of_widths => Set[int]
+    m = re.fullmatch(
+        rf"(?P<elems>\w*?)_?(?P<container>{containers})", name
+    ) or re.fullmatch(rf"(?P<container>{containers})_of_(?P<elems>\w*)", name)
+    if m:
+        # only do a simple container match
+        # and don't check all of BOOL_NAMES to not trigger on stuff like "save_list"
+        elems = m.group("elems")
+        for names, name_type in (
+            (("bool", "boolean"), "bool"),
+            # don't trigger on `real_list`
+            (FLOAT_NAMES - {"real"}, "float"),
+            (INTEGER_NAMES, "int"),
+            (STRING_NAMES | {"string", "str"}, "str"),
+        ):
+            if elems in names or (elems[-1] == "s" and elems[:-1] in names):
+                return NamedParam("", None, name_type), [
+                    m.group("container").capitalize()
+                ]
+
+    # Names which imply the value is a boolean
+    if name.startswith("is_") or name in BOOL_NAMES:
+        return NamedParam("", None, "bool"), []
+
+    if (
+        name.endswith("_size")
+        or (name.endswith("size") and "_" not in name)
+        or re.fullmatch(r"n(um)?_[a-z_]*s", name)
+        or name in INTEGER_NAMES
+    ):
+        return NamedParam("", None, "int"), []
+
+    if name in FLOAT_NAMES:
+        return NamedParam("", None, "float"), []
+
+    if (
+        "file" in name
+        or "path" in name
+        or name.endswith("_dir")
+        or name in ("fname", "dir", "dirname", "directory", "folder")
+    ):
+        # Common names for filesystem paths: these are usually strings, but we
+        # don't want to make strings more convenient than pathlib.Path.
+        return None, []
+
+    if (
+        name.endswith("_name")
+        or (name.endswith("name") and "_" not in name)
+        or ("string" in name and "as" not in name)
+        or name.endswith("label")
+        or name in STRING_NAMES
+    ):
+        return NamedParam("", None, "str"), []
+
+    # Last clever idea: maybe we're looking a plural, and know the singular:
+    # don't trigger on multiple ending "s" to avoid nested calls
+    if re.fullmatch(r"\w*[^s]s", name):
+        elems, container = guess_type_from_argname(name[:-1])
+        if elems is not None and not container:
+            return elems, ["List"]
```

Comment:
Shouldn't this often be an Iterable or Sequence or similar? Does Zac's data cover this case?

Reply:
Hmm, `List` is definitely the most common collection annotation, but I'm not convinced that makes it correct here.  Possibly a case where it makes sense to guess for Hypothesis, but `autotyping` would prefer something else?  If so I'd be inclined to go for either `Sequence` or not guessing at all here.

Reply:
I kinda like `Sequence`, it's easy to overspecify `List` (ime) even when you don't care about modifying it and you'll get a pretty obvious error if you do modify it. But depends on how vary of false positives you want to be.